src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/RunSelectedTestAction.java

```diff
@@ -0,0 +1,253 @@
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.facebook.buck.intellij.ideabuck.actions;
+
+import com.facebook.buck.intellij.ideabuck.build.BuckBuildManager;
+import com.facebook.buck.intellij.ideabuck.build.BuckCommand;
+import com.facebook.buck.intellij.ideabuck.build.BuckQueryCommandHandler;
+import com.facebook.buck.intellij.ideabuck.configurations.TestConfiguration;
+import com.facebook.buck.intellij.ideabuck.configurations.TestConfigurationType;
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.google.common.base.Function;
+import com.intellij.execution.Executor;
+import com.intellij.execution.ExecutorRegistry;
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.executors.DefaultDebugExecutor;
+import com.intellij.execution.impl.RunManagerImpl;
+import com.intellij.execution.impl.RunnerAndConfigurationSettingsImpl;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.LangDataKeys;
+import com.intellij.openapi.actionSystem.PlatformDataKeys;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Abstract class using common logic for running and debugging from the editor or project view.
+ */
+public abstract class RunSelectedTestAction extends AnAction {
+
+  private static final String ORG_JUNIT_TEST_ANNOTATION = "org.junit.Test";
+
+  @Override
+  public void update(final AnActionEvent event) {
+    super.update(event);
+
+    PsiElement element = event.getData(LangDataKeys.PSI_ELEMENT);
+    final Presentation presentation = event.getPresentation();
+
+    presentation.setEnabled(true);
+    if (element instanceof PsiMethod) {
+      PsiMethod method = (PsiMethod) element;
+      method.getModifierList().getAnnotations();
+      boolean isTestMethod;
+      isTestMethod = isTestMethod(method);
+      if (isTestMethod) {
```

Comment:
inline `isTestMethod(method)` directly and avoid the variable as you dont reuse it